src/kernels/MIOpenGLU.cpp

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef MIOPEN_DONT_USE_HIP_RUNTIME_HEADERS
+#include <hip/hip_fp16.h>
+#include <hip/hip_runtime.h>
+#endif
+
+#include "float_types.h"
+
+__device__ FLOAT_ACCUM sigmoid(FLOAT_ACCUM x) { return 1.0f / (1.0f + exp(-x)); }
+
+template <typename TI, typename TO>
+__device__ void GLUFwdCOntiguousKernel(const TI* __restrict__ inputFirstHalf,
```

Comment:
The kernel called GLU is basically an operator that receives input and output tensors one by one, receives dimention as a parameter.
And divides it in half based on dimention , then multiplies the first half of the divided input by the second half to which sigmoid is applied.
Since the work we do is creating replicas of pytorch or tensorflow, be sure to refer to the definitions of pytorch and tensorflow. (https://pytorch.org/docs/stable/generated/torch.nn.GLU.html) 
The API shared externally in miopen.h is used directly by users. Do not change the definition except in some unavoidable cases.
Therefore, in order to implement GLU in the same way as before, you must understand tensor view and slice. See https://github.com/ROCm/MIOpen/pull/2883. Please note that this PR is currently under review, so some details may change.